docs/core/integration-quality-scale/rules/iqs006.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "IQS006: Use ConfigEntry.runtime_data to store runtime data"
+---
+
+IQS006 is as follows:
+
+Use `ConfigEntry.runtime_data` to store runtime data
+
+## Reasoning
+
+The `ConfigEntry` object has a `runtime_data` attribute that can be used to store runtime data.
+This is useful for storing data that is not persisted to the configuration file, but is needed during the lifetime of the configuration entry.
+
+By using `runtime_data`, we maintain consistency for developers to store runtime data in a consistent and typed way.
+Because of the added typing, we can use tooling to avoid typing mistakes.
+
+## Example implementation
+
+The type of a `ConfigEntry` can be extended with the type of the data put in `runtime_data`.
+In the following example, we extend the `ConfigEntry` type with `MyClient`, which means that the `runtime_data` attribute will be of type `MyClient`.
+
+```python
+
+`__init__.py`:
+```python
+type MyIntegrationConfigEntry = ConfigEntry[MyClient]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: MyIntegrationConfigEntry) -> bool:
+    """Set up my intergation from a config entry."""
+
+    client = MyClient(entry.data[CONF_HOST])
+
+    entry.runtime_data = client
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+```
+
+## Additional resources
+
+More information about config entries and their lifecycle can be found in the [config entry documentation](../../../config_entries_index).
```

Comment:
We're mixing the use of "config entry" and "configuration entry". Let's pick one and stick to that in general for these rules.

Reply:
Any preference? I would go for configuration entry

Reply:
Yes, I agree.